core/src/types/blocking_read/blocking_reader.rs

```diff
@@ -74,24 +84,26 @@ impl BlockingReader {
             }
         }
 
+        let iter = BufferIterator::new(
+            self.inner.clone(),
+            self.options.chunk(),
```

Comment:
We can accpet an `OpReader` directly.

Reply:
> Would you like to submit this PR first?

Do you mean submit a new PR to change oio::BlockingReader to Arc<dyn oio::BlockingRead> first?

Reply:
> Do you mean submit a new PR to change oio::BlockingReader to Arc first?

Yes